util/EIP-712/src/eip712.rs

```diff
@@ -175,4 +186,43 @@ mod tests {
 		let data = from_str::<EIP712>(string).unwrap();
 		assert_eq!(data.validate().is_err(), true);
 	}
+
+	#[test]
+	fn test_invalid_domain() {
```

Comment:
```suggestion
	fn domain_needs_at_least_one_field() {
```

Reply:
Dare I ask for a test with a valid domain too? :)

Reply:
Done!